viewsb/analyzer.py

```diff
@@ -57,13 +57,27 @@ def __init__(self, backend, frontend, decoders=None):
         # Create our analysis queue.
         self.analysis_queue = queue.Queue()
 
+        # Create our exception pipes
+        self._in_backend_except_conn, self._out_backend_except_conn = multiprocessing.Pipe()
+        self._in_frontend_except_conn, self._out_frontend_except_conn = multiprocessing.Pipe()
```

Comment:
If I'm understanding this correctly, currently `_in_frontend_except_conn` and `_out_frontend_except_conn` are both unused, right? `_out_frontend_except_conn`—a pipe theoretically for the backend to receive exceptions from the frontend—isn't even passed to the backend, and, unless I'm missing something, its peer, `_in_frontend_except_conn`, isn't used in the frontend class it's passed to. If they're entirely unused, do we need that second line at all?

---

Also, when writing that comment I had to go through some mental gymnastics to make sure I was talking about the right pipe/connection. I ended up creating these comments above those definitions so I could keep track:
```python
# _in_backend_except_conn: backend writes exceptions to frontend
# _out_backend_except_conn: frontend reads exceptions from backend

# _in_frontend_except_conn: frontend writes exceptions to backend
# _out_frontend_except_conn: backend reads exceptions from frontend
```
Could we rename them to something more clear? I thought of something like `pipe_send_except_to_frontend`, `pipe_recv_except_from_frontend`, et cetera, which while verbose is clear enough to be fine, but I'm definitely open to other suggestions. Also might be nice to name them something more meaningful on the `ProcessManager` side as well, but since that class is generic to both backends and frontends I can't think of much other than something like `pipe_send_except`.

Reply:
Kinda, `_in_frontend_except_conn` is passed to the frontend to send the exceptions, but we never handle them. I mean, the exceptions are thrown and the process finishes, but we never do anything with their information. I was planning on handling the frontend exceptions in another place, but didn't want to introduce more stuff into this PR. I think we should definitely print the frontend exceptions to console.

The names don't make much difference to me, but I can definitely agree they are not super clear. Your suggested names seem fine to me, we should change.